File: src/dlp/ast/sentences/Log.ts

import {Expression} from "../expressions/Expression";
import {AbstractLocatable} from "../AbstractLocatable";
import {Sentence} from "./Sentence";

export class Assignment extends AbstractLocatable implements Sentence {

    expression: Expression;

    constructor( expression: Expression, line: number, column: number){
        super(line, column);
        this.expression = expression;
    }
}

File: src/ast/sentences/Log.java
package ast.sentences;

import ast.AbstractLocatable;
import ast.expressions.Expression;

public class Log extends AbstractLocatable implements Sentence {

    private Expression expression;

    public Log(Expression expression, int line, int column) {
        super(line, column);
        this.expression = expression;
    }

    public Expression getExpression() {
        return expression;
    }
}
